manager/allocator/network.go

```diff
@@ -641,6 +644,35 @@ func (a *Allocator) commitAllocatedNode(ctx context.Context, batch *store.Batch,
 	return nil
 }
 
+func updatePortsInHostPublishMode(s *api.Service) {
```

Comment:
I agree in principle, but this would involve moving it to the `api` package, and there was a decision against adding non-generated code in this package. So for now we're unable to define methods on `api` types.